test/server/dom-shims.js

```diff
@@ -41,3 +41,39 @@ describe(`customElement with shadowDom`, function () {
     expect(elem.el.innerHTML).to.equal(`<style>color: blue;</style><div class="foo"><p>Hello</p></div>`);
   });
 });
+
+describe(`customElement callbacks`, function () {
+  it(`calls connectedCallback and disconnectedCallback when adding/removing from DOM`, async function () {
+    const elem = document.createElement(`my-app`);
+    const connectedCallbackSpy = sinon.spy(elem, `connectedCallback`);
+    const disconnectedCallbackSpy = sinon.spy(elem, `disconnectedCallback`);
+
+    expect(elem.isConnected).to.not.exist;
+
+    // fake adding to DOM
+    const parent = document.createElement(`div`);
+    parent.appendChild(elem);
+
+    await nextAnimationFrame();
+    expect(connectedCallbackSpy.called).to.be.true;
```

Comment:
since you've specified sinon-chai as a top-level dep now, may as well turn it on in this module to get assertions like `expect(connectedCallbackSpy).to.have.been.called;`